### PR TITLE
Azure migration should convert "| Microsoft Azure" to "| Microsoft Docs" in title

### PR DESCRIPTION
**Pull request by jsntcy**
Azure migration should convert "| Microsoft Azure" to "| Microsoft Docs" in title

### CODE REVIEW

File: src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs [hunk 1]
```diff
@@ -82,7 +82,13 @@ private AzureHtmlMetadata GetAttributesFromHtmlContent(string htmlContent)
                 {
                     if (string.Equals(attribute.Name, "pageTitle", StringComparison.OrdinalIgnoreCase))
                     {
-                        attributes["title"] = attribute.Value;
+                        // Per Azure's request, migration script should convert "| Microsoft Azure" to "| Microsoft Docs" in title.
+                        var title = attribute.Value;
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs]: >title [](start = 28, length = 5)

should we also update test TestAzureMarkdownRewriters_AzureProperties?

File: src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs [hunk 2]
```diff
@@ -82,7 +82,13 @@ private AzureHtmlMetadata GetAttributesFromHtmlContent(string htmlContent)
                 {
                     if (string.Equals(attribute.Name, "pageTitle", StringComparison.OrdinalIgnoreCase))
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs]: >pageTitle [](start = 55, length = 9)

should we also replace "pageTitle" itself ?

File: src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs [hunk 3]
```diff
@@ -82,7 +82,13 @@ private AzureHtmlMetadata GetAttributesFromHtmlContent(string htmlContent)
...
                     {
-                        attributes["title"] = attribute.Value;
+                        // Per Azure's request, migration script should convert "| Microsoft Azure" to "| Microsoft Docs" in title.
+                        var title = attribute.Value;
+                        if (title != null && title.Contains("| Microsoft Azure"))
```
Comment (Huachao) [inline src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs]: >&& title.Contains("| Microsoft Azure") [](start = 42, length = 38)

can we directly replace without checking whether contains

**Comment by hellosnow** [comment]
>         private Dictionary<string, string> GetAttributesFromNode(HtmlNode node)

not sure whether AzureHtmlMetadataBlockRule needs to be updated?

---
Refers to: src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs:76 in 930ccec. [](commit_id = 930ccecda901eb4ea13bc362a606cb784a2c3004, deletion_comment = False)

File: src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs (hunk 2 above)
Comment (jsntcy) [inline src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs]: "pageTitle" has already been converted to "title".

---
In reply to: [85067936](https://github.com/dotnet/docfx/pull/826#discussion_r85067936) [](ancestors = 85067936)

**Comment by jsntcy** [comment]
>         private Dictionary<string, string> GetAttributesFromNode(HtmlNode node)

It looks like we don't need to update AzureHtmlMetadataBlockRule since migration tool only call AzureMigrationHtmlMetadataBlockRule.
Wait for @feng xu 's confirmation.

---
In reply to: [256277183](https://github.com/dotnet/docfx/pull/826#issuecomment-256277183) [](ancestors = 256277183)

---
Refers to: src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs:76 in 930ccec. [](commit_id = 930ccecda901eb4ea13bc362a606cb784a2c3004, deletion_comment = False)

File: src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs (hunk 1 above)
Comment (jsntcy) [inline src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs]: No, TestAzureMarkdownRewriters_AzureProperties only test AzureHtmlMetadataBlockRule 

---
In reply to: [85067711](https://github.com/dotnet/docfx/pull/826#discussion_r85067711) [](ancestors = 85067711)

**Comment by jsntcy** [comment]
>         private Dictionary<string, string> GetAttributesFromNode(HtmlNode node)

Confirmed with Feng that we don't need update AzureHtmlMetadataBlockRule 

---
In reply to: [256281599](https://github.com/dotnet/docfx/pull/826#issuecomment-256281599) [](ancestors = 256281599,256277183)

---
Refers to: src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs:76 in 930ccec. [](commit_id = 930ccecda901eb4ea13bc362a606cb784a2c3004, deletion_comment = False)

File: src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs [hunk 4]
```diff
@@ -82,7 +82,13 @@ private AzureHtmlMetadata GetAttributesFromHtmlContent(string htmlContent)
                 {
                     if (string.Equals(attribute.Name, "pageTitle", StringComparison.OrdinalIgnoreCase))
                     {
-                        attributes["title"] = attribute.Value;
+                        // Per Azure's request, migration script should convert "| Microsoft Azure" at the end of title to "| Microsoft Docs".
+                        var title = attribute.Value;
+                        if (title != null && title.EndsWith("| Microsoft Azure"))
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs]: >title [](start = 45, length = 5)

is it necessary to trim?

File: src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs [hunk 5]
```diff
@@ -82,7 +82,13 @@ private AzureHtmlMetadata GetAttributesFromHtmlContent(string htmlContent)
...
+                        {
+                            title = title.Replace("| Microsoft Azure", "| Microsoft Docs");
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs]: >title [](start = 36, length = 5)

how about the case that "| Microsoft Azure abc | Microsoft Azure"?

Comment (jsntcy) [inline src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs]: Use regular expression now.

---
In reply to: [85080171](https://github.com/dotnet/docfx/pull/826#discussion_r85080171) [](ancestors = 85080171)

**Comment by Huachao** [comment]
:shipit:

File: src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs (hunk 3 above)
Comment (Huachao) [inline src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureBlockRules/AzureMigrationHtmlMetadataBlockRule.cs]: @fenxu you are reviewing an old version

**Pull request merged**